### ISSUE TITLE
Allow multiple breakouts of the same date/time column with different granularities

### ISSUE THREAD

**Issue by christopheg** [metabase/metabase#4726]
Hi,

I don't know how to classify this request. Maybe what I'm looking for is already possible and I just need someone helping me out here. If not, it could be a feature request.

The 'problem' is the following: Image a table (ex 'invoice') which has the following fields:
'id', 'created', 'price'
I would like to produce a line chart with X = 'months of year' and Y = 'total value'.
To make the values comparable per year, I would like to have a line per year.
Each line would than plot the total amount invoiced for a month in the given year.
I've found a comparable chart which shows what I'm looking for:
https://harpoonapp.com/static/uploads/docs-dashboard-revenue-per-month.png

Would this be possible?

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by tlrobinson** [comment]
This is *almost* possible, except that we only allow you to break out by each column once, even if you were to pick a different date granularity. You'd do basically "sum of price, grouped by created: month-of-year, created: year".

I'm kind of faking it here by using a different date column for the two breakouts:

![screenshot 2017-04-07 12 55 12](https://cloud.githubusercontent.com/assets/18193/24817475/75054762-1b91-11e7-9fb7-72190f4ace89.png)

**Comment by tlrobinson** [comment]
I'd propose removing the uniqueness requirement for date/time breakouts. @metabase/core-developers ?

**Comment by christopheg** [comment]
I love to see this feature!

**Comment by atharvai** [comment]
is this going to be fixed in the next version (0.25 or 0.24.3)? This was working in 0.23.x but not in 0.24.x

**Comment by salsakran** [comment]
@camsaul is this also a limitation in MBQL or just a frontend issue?

**Comment by camsaul** [comment]
> This is almost possible, except that we only allow you to break out by each column once

@salsakran I'm fairly certain this is only a QB/frontend constraint and if we tweak that code the query should work again, e.g.

> This was working in 0.23.x but not in 0.24.x

**Comment by salsakran** [comment]
@camsaul 
From the WIP PR linked above -- 
"If you group by the same date field multiple times with different granularities, the backend response lacks id, table_id, unit and some other properties for the column of second breakout. Frontend is unable to display the column correctly because of that. The dataset query body for reproducing this:"

**Comment by camsaul** [comment]
@christopheg are you still seeing this in Metabase 0.31.1? This might have been fixed already

**Comment by cproctor1988** [comment]
Currently running v. 32.10 and still seeing this issue. 
![image](https://user-images.githubusercontent.com/33880391/63376384-89e11c00-c342-11e9-8c01-e32c09c6383f.png)

**Comment by eljakim** [comment]
@camsaul I see that the milestone for this ticket just keeps being postponed. Is there a realistic timeframe if/when this will be implemented?

**Comment by mijamo** [comment]
Interestingly it works when you use a field from a previous question, but not for a field from a database. So in the meantime a workaround is to create a question that simply duplicates a table (basically just select the table and save the question) and then create a nested question based on the duplication.

**Comment by eljakim** [comment]
We were quite eager to start using Metabase, but the speed of resolving this issue has caused us to give up. The milestone moving does not really help, and I apologise for asking @camsaul for his plan, but then not responding, and *again* milestone moving without any comments is just silly.

**Comment by EBoisseauSierra** [comment]
Well… I almost raised an issue when I found this one. Here is, for the record, how I formulated it.

## Background

I want to see my "Year on Year" revenue. I can do this as a SQL query as follows:

![1_shadow](https://user-images.githubusercontent.com/37387755/74462367-05ea9580-4e88-11ea-80f7-879c29d703d4.png)


I would, however, be able to do this via the GUI as a “custom question”. Yet at the moment, I can only aggregate the `sum(Total)` by:
* `Created At: Month`:  
![2_shadow](https://user-images.githubusercontent.com/37387755/74462379-0a16b300-4e88-11ea-91b9-8647bdf6b848.png)
… but this gives me a _continuous_ list of year-month values (i.e. not able to have values of February 2019 and February 202 on the same x value)  
![3_shadow](https://user-images.githubusercontent.com/37387755/74462388-0f73fd80-4e88-11ea-920d-f9b502d7e308.png)
* `Created At: Month of the Year`  
![4_shadow](https://user-images.githubusercontent.com/37387755/74462405-14d14800-4e88-11ea-85e7-f7bc18a8bd02.png)
… but this gives me the sales _aggregated over all years_ for a given month:  
![5_shadow](https://user-images.githubusercontent.com/37387755/74462412-18fd6580-4e88-11ea-933c-df820f979226.png)

And if I aggregate by `Created At: Year`, I cannot add an additional aggregate on the `Created At` column.  
![6_shadow](https://user-images.githubusercontent.com/37387755/74462423-1dc21980-4e88-11ea-8ad8-b230519434ff.png)

Yet I would like indeed to be able to aggregate by  `Created At: Year` _and_  `Created At: Month of the Year`, so I get result of the form:

| Year | Month of the year | Sales |
| ---: | ---: | ---: |
| 2019 | 01 | 1234 |
| 2019 | 02 | 2345 |
| … | … | … |
| 2020 | 01 | 1235 |
| 2020 | 02 | 42 |

## Expected solution

I want to be able to get a result of the form above, potentially by being able to aggregate my Metric by `Created At: Year` _and_  `Created At: Month of the Year`

## Alternative
The SQL is straightforward, but I would prefer using the GUI.

## Severity
It's prevents non SQL-savvy user to create "Year on Year" charts. But I'd say it's rather a nice-to-have. 

## Context
Metabase v0.34.0

**Comment by magnusfagerlund** [comment]
I see that this has got the 0.35 milestone. Will it be included in this version?

**Comment by eljakim** [comment]
I don't think this will ever be included.

@camsaul would it be possible for you to respond to us? You keep changing pushing it farther back, but you have not yet responded any time.

**Comment by flamber** [comment]
@eljakim It's more complicated than just removing uniqueness from the fields. We would need to handle this correctly for nested queries in general, so there isn't a conflict with duplicate fields. It's definitely something we want to be able to do.
The milestone changes were done in batches, so every issue that we tried to accomplish, but didn't complete was moved to the next milestone. We are now trying to set more realistic goals for each milestone. There are 2000 other issues, and we'll try to fix every issue, but we're a small team.
Related to #11371
